cocos/3d/misc/mesh-codec.ts

```diff
@@ -63,31 +63,32 @@ function initDecoderWASM (wasm_factory: any, wasm_url: string): Promise<void> {
 }
 
 function shouldUseWasmModule (): boolean {
-    if (WASM_SUPPORT_MODE === (WebAssemblySupportMode.MAYBE_SUPPORT as number)) {
+    if (NATIVE_CODE_BUNDLE_MODE === (NativeCodeBundleMode.BOTH as number)) {
         return sys.hasFeature(sys.Feature.WASM);
-    } else if (WASM_SUPPORT_MODE === (WebAssemblySupportMode.SUPPORT as number)) {
+    } else if (NATIVE_CODE_BUNDLE_MODE === (NativeCodeBundleMode.WASM as number)) {
         return true;
     } else {
         return false;
     }
 }
 
 export function InitDecoder (): Promise<void> {
-    return ensureWasmModuleReady().then(() => Promise.all([
-        import('external:emscripten/meshopt/meshopt_decoder.asm.js'),
-        import('external:emscripten/meshopt/meshopt_decoder.wasm.js'),
-        import('external:emscripten/meshopt/meshopt_decoder.wasm.wasm'),
-    ]).then(([
-        { default: meshopt_asm_factory },
-        { default: meshopt_wasm_factory },
-        { default: meshopt_wasm_url },
-    ]) => {
-        if (shouldUseWasmModule() && meshopt_wasm_url) {
-            return initDecoderWASM(meshopt_wasm_factory, meshopt_wasm_url);
+    const errorReport = (msg: any): void => { error(msg); };
+    return ensureWasmModuleReady().then(() => {
+        if (shouldUseWasmModule()) {
+            return Promise.all([
+                import('external:emscripten/meshopt/meshopt_decoder.wasm.js'),
+                import('external:emscripten/meshopt/meshopt_decoder.wasm.wasm'),
+            ]).then(([
+                { default: meshopt_wasm_factory },
+                { default: meshopt_wasm_url },
+            ]) => initDecoderWASM(meshopt_wasm_factory, meshopt_wasm_url));
         } else {
-            return initDecoderASM(meshopt_asm_factory);
+            return import('external:emscripten/meshopt/meshopt_decoder.asm.js').then(
+                ({ default: meshopt_asm_factory }) => initDecoderASM(meshopt_asm_factory),
+            );
         }
-    }));
+    }).catch(errorReport);
```

Comment:
Is it better to print the error log?

Reply:
Yes, `errorReport` is a function for print error log.
```ts
 const errorReport = (msg: any): void => { error(msg); };
```
It's the old code written like this.